File: avalon-application/avalon-standalone-server/src/main/java/io/github/lishangbu/avalon/standalone/AvalonStandaloneApplication.java
package io.github.lishangbu.avalon.standalone;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.persistence.autoconfigure.EntityScan;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.data.web.config.EnableSpringDataWebSupport;

@EnableSpringDataWebSupport
@EnableJpaRepositories("io.github.lishangbu.avalon.**.repository")
@EntityScan("io.github.lishangbu.avalon.**.entity")
@SpringBootApplication(scanBasePackages = "io.github.lishangbu.avalon")
public class AvalonStandaloneApplication {

  static void main(String[] args) {
    SpringApplication.run(AvalonStandaloneApplication.class, args);
  }
}

File: avalon-application/avalon-standalone-server/src/main/kotlin/io/github/lishangbu/avalon/standalone/AvalonStandaloneApplication.kt
package io.github.lishangbu.avalon.standalone

import org.babyfish.jimmer.spring.repository.EnableJimmerRepositories
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication

/**
 * 单体应用启动类
 *
 * Spring Boot 单体应用的入口
 *
 * @author lishangbu
 * @since 2025/8/24
 */
@EnableJimmerRepositories("io.github.lishangbu.avalon.**.repository")
@SpringBootApplication(scanBasePackages = ["io.github.lishangbu.avalon"])
class AvalonStandaloneApplication

fun main(args: Array<String>) {
    runApplication<AvalonStandaloneApplication>(*args)
}
